storage/libradb/src/ledger_store/ledger_info_test.rs

```diff
@@ -6,67 +6,133 @@ use crate::{change_set::ChangeSet, LibraDB};
 use libra_tools::tempdir::TempPath;
 use libra_types::block_info::BlockInfo;
 use libra_types::ledger_info::LedgerInfo;
+use libra_types::validator_set::ValidatorSet;
 use proptest::{collection::vec, prelude::*};
+use std::collections::BTreeMap;
 
-prop_compose! {
-    fn arb_partial_ledger_info()(accu_hash in any::<HashValue>(),
-                                 consensus_hash in any::<HashValue>(),
-                                 timestamp in any::<u64>()) -> (HashValue, HashValue, u64) {
-        (accu_hash, consensus_hash, timestamp)
-    }
-}
-
-prop_compose! {
-    fn arb_ledger_infos_with_sigs()(
-        partial_ledger_infos_with_sigs in vec(
-            any_with::<LedgerInfoWithSignatures>((1..3).into()).no_shrink(), 1..100
+fn arb_ledger_infos_with_sigs() -> impl Strategy<Value = Vec<(LedgerInfoWithSignatures, bool)>> {
+    (
+        vec(
+            (
+                any::<HashValue>(),
+                any::<HashValue>(),
+                any::<HashValue>(),
+                any::<u64>(),
+                0..10u64,
+                any::<bool>(),
+            ),
+            1..100,
         ),
-        start_epoch in 0..10000u64,
-    ) -> Vec<LedgerInfoWithSignatures> {
-        partial_ledger_infos_with_sigs
-            .iter()
-            .enumerate()
-            .map(|(i, p)| {
-                let ledger_info = p.ledger_info();
-                LedgerInfoWithSignatures::new(
-                    LedgerInfo::new(
-                        BlockInfo::new(
-                            start_epoch + i as u64,
-                            start_epoch + i as u64,
-                            ledger_info.consensus_block_id(),
-                            ledger_info.transaction_accumulator_hash(),
-                            start_epoch + i as Version,
-                            ledger_info.timestamp_usecs(),
-                            None,
-                        ),
-                        ledger_info.consensus_data_hash(),
-                    ),
-                    p.signatures().clone(),
+        vec(0..100_000_000u64, 3),
+    )
+        .prop_map(|(block_params, mut ledger_params)| {
+            ledger_params.sort_unstable();
+            let mut epoch = ledger_params[0];
+            let mut round = ledger_params[1];
+            let mut version = ledger_params[2];
```

Comment:
From the code below I feel you don't have the `round <= version` constraint. Also I think it might be more intuitive to just let these parameters to start from 0 and get rid of this random `vec`.

Reply:
I think let it start in the middle of the time may reveal some conner case bugs. Maybe not worth it if there's too much confusion reading it. 
But I'm refactoring this out in the coming diff... so will leave it alone for now.